components/count-badge/count-badge-mixin.js

```diff
@@ -0,0 +1,196 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../tooltip/tooltip.js';
+import { css, html } from 'lit-element/lit-element.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map';
+
+export const countBadgeStyles = css`
+:host([hidden]) {
+	display: none;
+}
+
+:host {
+	display: inline-block;
+	min-width: 0.9rem;
+}
+
+.d2l-count-badge-number {
+	border: 2px solid white;
+	font-weight: bold;
+}
+
+:host([type="notification"]) .d2l-count-badge-number {
+	background-color: var(--d2l-color-carnelian-minus-1);
+	color: white;
+}
+
+:host([type="count"]) .d2l-count-badge-number {
+	background-color: var(--d2l-color-gypsum);
+	color: var(--d2l-color-tungsten);
+}
+
+:host([size="small"]) .d2l-count-badge-number {
+	border-radius: 0.55rem;
+	font-size: 0.6rem;
+	line-height: 0.9rem;
+	padding-left: 0.3rem;
+	padding-right: 0.3rem;
+}
+
+:host([size="large"]) .d2l-count-badge-number {
+	border-radius: 0.7rem;
+	font-size: 0.8rem;
+	line-height: 1.2rem;
+	padding-left: 0.4rem;
+	padding-right: 0.4rem;
+}
+
+:host([size="small"]) .d2l-count-badge-wrapper {
+	border-radius: 0.65rem;
+	outline: none;
+}
+
+:host([size="large"]) .d2l-count-badge-wrapper {
+	border-radius: 0.8rem;
+	outline: none;
+}
+`;
+
+export const CountBadgeMixin = superclass => class extends LocalizeCoreElement(RtlMixin(superclass)) {
```

Comment:
All the logic that was previously in the count-badge has been extracted to this Mixin, so we have less duplicated code. There are a few changes that I'll point out below in the render function.